lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,173 @@
+
+        {
+
+        }
+        """
+        logger = self.app.logger
+        elasticsearch = f"{self.app.config_elasticsearch.get('host')}:{self.app.config_elasticsearch.get('port')}"
+        json_data = request.get_json(silent=True)
+        if not json_data:
+            logger.error(
+                "QueryControllers: Invalid json object. Query: {}", request.url
+            )
+            abort(400, message="QueryControllers: Invalid json object in request")
+
+        user = json_data["user"]
+        prefix = json_data["prefix"]
+        start = json_data["start"]
+        end = json_data["end"]
+        session_token = request.headers.get("X-auth-token")
+
+        # We have nothing to authorize against yet, but print the specified
+        # user and session token to prove we got them (and so the variables
+        # aren't diagnosed as "unused")
+        logger.info(
+            "QueryControllers POST for user {}, prefix {}, session {}: ({} - {})",
+            user,
+            prefix,
+            session_token,
+            start,
+            end,
+        )
+
+        # TODO: we could make this a constant template, then copy it and
+        # replace the start and end times. Is that worthwhile?
+        payload = {
+            "query": {"range": {"@timestamp": {"gte": start, "lte": end}}},
+            "aggs": {
+                "controllers": {
+                    "terms": {"field": "controller", "order": [{"runs": "desc"}]},
+                    "aggs": {"runs": {"max": {"field": "run.start"}}},
+                }
+            },
+        }
+
+        uri_fragment = QueryControllers._genMonthUriRange(
+            prefix, "v6.run-data.", start, end
+        )
+
+        try:
+            # query Elasticsearch
+            es_response = requests.post(
+                f"{elasticsearch}/{uri_fragment}/_search",
+                params={"ignore_unavailable": "true"},
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+                json=payload,
+            )
+        except requests.exceptions.ConnectionError:
+            logger.exception("Connection refused during the Elasticsearch post request")
+            abort(
+                500, message="Network problem, could not post to Elasticsearch Endpoint"
+            )
+        except requests.exceptions.Timeout:
+            logger.exception(
+                "Connection timed out during the Elasticsearch post request"
+            )
+            abort(
+                500,
+                message="Connection timed out, could not post to Elasticsearch Endpoint",
+            )
+        except requests.exceptions.InvalidURL:
+            logger.exception("Invalid url during the Elasticsearch post request")
+            abort(
+                500,
+                message="Invalid Elasticsearch url, could not complete the post request",
+            )
```

Comment:
Minor, but we can add `HTTPError` exception as well if the HTTP request to Elasticsearch returned an unsuccessful status code [requests exceptions](https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions)

Reply:
I stole most of this from the existing Elasticsearch query; but that's probably a good idea.